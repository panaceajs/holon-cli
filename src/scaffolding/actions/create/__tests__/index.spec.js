const { resolve } = require('path');
const snapshot = require('../../../../utils/task-snapshot');
const task = require('../');

describe('actions', () => {
  it('should create actions in `actions/index.js`', async () => {
    await expect(
      snapshot(task, {
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
  it('should create actions in `other-dir/actions/index.js`', async () => {
    await expect(
      snapshot(task, {
        cwd: resolve('other-dir'),
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
});
