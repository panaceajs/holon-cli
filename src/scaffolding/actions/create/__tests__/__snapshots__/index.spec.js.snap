// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create actions in \`actions/index.js\` 1`] = `
"
--------------------------------------------------------------------------------
// ./actions/index.js
// Create
--------------------------------------------------------------------------------

import { createAction } from 'redux-actions';
import { AN_ACTION, ANOTHER_ACTION } from '../action-types';

export const anAction = createAction(AN_ACTION);
export const anotherAction = createAction(ANOTHER_ACTION);





"
`;

exports[`actions should create actions in \`other-dir/actions/index.js\` 1`] = `
"
--------------------------------------------------------------------------------
// ./actions/index.js
// Create
--------------------------------------------------------------------------------

import { createAction } from 'redux-actions';
import { AN_ACTION, ANOTHER_ACTION } from '../action-types';

export const anAction = createAction(AN_ACTION);
export const anotherAction = createAction(ANOTHER_ACTION);





"
`;
